database_wrapper.py

```diff
@@ -486,126 +524,66 @@ def save_question_answer_pair(self, qa_dict: dict) -> bool:
-        self.session.commit()
-        return True
+        # Validate input
+        self.validate_input_keys(data_dict, EXPECTED_KEYS_BY_ENTITY[entity_type])
 
-    def save_club(self, formatted_data: dict) -> bool:
-        """
-        Save the given club into the database.
-
-         Example input:
-         {
-             "club_name": Cal Poly Computer Science and Artificial Intelligence
-             "types": Academic, Special Interest
-             "desc": The Computer Science and Artificial Intelligence club provides..."
-             "contact_email": maikens@calpoly.edu
-             "contact_email_2": fkurfess@calpoly.edu
-             "contact_person": Miles Aikens
-             "contact_phone": 7349723564
-             "box": 89
-             "advisor": Franz Kurfess
-             "affiliation": None
-         }
-
-         Args:
-             formatted_data: a dictionary that corresponds to the fields in Clubs
-
-         Raises:
-             BadDictionaryKeyError - ...
-             BadDictionaryValueError - ...
-
-         Returns:
-             True if all is good, else False
-        """
+        # If the data is supposed to be formatted, format it!
+        if entity_type in format_method_by_entity:
+            data_dict = format_method_by_entity[entity_type](data_dict)
 
-        expected_keys = {'club_name', 'types', 'desc', 'contact_email',
-                         'contact_email_2', 'contact_person', 'contact_phone',
-                         'box', 'advisor', 'affiliation'}
-        self.validate_input_keys(formatted_data, expected_keys)
-
-        club_data = self.session.query(Clubs).filter_by(club_name=formatted_data['club_name']).first()
-        if not club_data:
-            club_data = Clubs()
-
-        club_data.club_name = formatted_data['club_name']
-        club_data.types = formatted_data['types']
-        club_data.desc = formatted_data['desc']
-        club_data.contact_email = formatted_data['contact_email']
-        club_data.contact_email_2 = formatted_data['contact_email_2']
-        club_data.contact_person = formatted_data['contact_person']
-        club_data.contact_phone = formatted_data['contact_phone']
-        club_data.box = formatted_data['box']
-        club_data.advisor = formatted_data['advisor']
-        club_data.affiliation = formatted_data['affiliation']
-
-        self.session.add(club_data)
+        # Grab the entity class attributes and initialize entity to None
+        entity_attributes = entity_type.__dict__
+        entity = None
+
+        # If filter_fields is provided, we can check if we are performing an UPDATE or INSERT.
+        if not filter_fields == []:
+            query = self.session.query(entity_type)
+
+            for field in filter_fields:
+                query = query.filter(getattr(entity_type, field) == data_dict[field])
+
+            entity = query.first()
+            if entity:
+                print("\033[93mUpdating {} in {}...\033[00m".format(
```

Comment:
let's consider putting the color codes into a GLOBAL_VARIABLE that would allow for
1. readabilitiy
2. easier changing of the colors